### DISCUSSION TITLE
can the idea of ZeroSync be used for ethereum?

### DISCUSSION THREAD

**Discussion by hustnn** [ZeroSync/ZeroSync#35] [Q&A] [answered]
See the question in title.

**Comment by lucidLuckylee** [comment, marked as answer]
I am aware of [fossil](https://github.com/OilerNetwork/fossil) and [herodotus](https://twitter.com/HerodotusDev?s=20&t=EvnvCUbO5x7QCOvcYFdPlw) that prove the (maybe partial) correctness of the hash chain for Ethereum. Not sure if there is anything holding one back from proving the transactions too other than the huge computational cost.

**Reply by hustnn** [reply]
thanks

**Comment by marcellobardus** [comment]
Hey yes, you can checkout this repo https://github.com/HerodotusDev/herodotus-eth-starknet
We will soon have a pure cairo implementation

**Reply by hustnn** [reply]
cool, looking forward to it.
